### ISSUE TITLE
`minikube tunnel` bind address flag doesn't work on Linux

### ISSUE THREAD

**Issue by hal-shin** [kubernetes/minikube#14712]
### What Happened?

The ability to specify a bind address was added in https://github.com/kubernetes/minikube/pull/14245 but it doesn't seem to work on Linux (Ubuntu) while it works on Windows.

Not sure if it's an SSH issue, but the SSH version is 8.2.

### Attach the log file

$ minikube tunnel --bind-address=127.0.0.1 --alsologtostderr
I0803 02:36:33.295925   16878 out.go:296] Setting OutFile to fd 1 ...
I0803 02:36:33.296078   16878 out.go:348] isatty.IsTerminal(1) = true
I0803 02:36:33.296086   16878 out.go:309] Setting ErrFile to fd 2...
I0803 02:36:33.296095   16878 out.go:348] isatty.IsTerminal(2) = true
I0803 02:36:33.296256   16878 root.go:329] Updating PATH: /home/akash/.minikube/bin
I0803 02:36:33.296398   16878 mustload.go:65] Loading cluster: granular
I0803 02:36:33.296677   16878 config.go:178] Loaded profile config "granular": Driver=docker, ContainerRuntime=docker, KubernetesVersion=v1.23.1
I0803 02:36:33.297004   16878 cli_runner.go:164] Run: docker container inspect granular --format={{.State.Status}}
I0803 02:36:33.335247   16878 host.go:66] Checking if "granular" exists ...
I0803 02:36:33.335619   16878 cli_runner.go:164] Run: docker system info --format "{{json .}}"
I0803 02:36:33.487439   16878 info.go:265] docker info: {ID:Q5VS:AXS5:5OT5:TY65:5U54:ASPN:KYBK:NIL6:A4QG:ELCP:G4ND:LQGS Containers:5 ContainersRunning:1 ContainersPaused:0 ContainersStopped:4 Images:45 Driver:overlay2 DriverStatus:[[Backing Filesystem extfs] [Supports d_type true] [Native Overlay Diff true] [userxattr false]] SystemStatus:<nil> Plugins:{Volume:[local] Network:[bridge host ipvlan macvlan null overlay] Authorization:<nil> Log:[awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog]} MemoryLimit:true SwapLimit:true KernelMemory:true KernelMemoryTCP:true CPUCfsPeriod:true CPUCfsQuota:true CPUShares:true CPUSet:true PidsLimit:true IPv4Forwarding:true BridgeNfIptables:true BridgeNfIP6Tables:true Debug:false NFd:32 OomKillDisable:true NGoroutines:39 SystemTime:2022-08-03 02:36:33.374834314 +0530 IST LoggingDriver:json-file CgroupDriver:cgroupfs NEventsListener:0 KernelVersion:5.15.0-41-generic OperatingSystem:Ubuntu 20.04.4 LTS OSType:linux Architecture:x86_64 IndexServerAddress:https://index.docker.io/v1/ RegistryConfig:{AllowNondistributableArtifactsCIDRs:[] AllowNondistributableArtifactsHostnames:[] InsecureRegistryCIDRs:[127.0.0.0/8] IndexConfigs:{DockerIo:{Name:docker.io Mirrors:[] Secure:true Official:true}} Mirrors:[]} NCPU:16 MemTotal:16164229120 GenericResources:<nil> DockerRootDir:/var/lib/docker HTTPProxy: HTTPSProxy: NoProxy: Name:akash-ASUS-TUF-Gaming-A15-FA506IC-FA506IC Labels:[] ExperimentalBuild:false ServerVersion:20.10.17 ClusterStore: ClusterAdvertise: Runtimes:{Runc:{Path:runc}} DefaultRuntime:runc Swarm:{NodeID: NodeAddr: LocalNodeState:inactive ControlAvailable:false Error: RemoteManagers:<nil>} LiveRestoreEnabled:false Isolation: InitBinary:docker-init ContainerdCommit:{ID:10c12954828e7c7c9b6e0ea9b0c02b01407d3ae1 Expected:10c12954828e7c7c9b6e0ea9b0c02b01407d3ae1} RuncCommit:{ID:v1.1.2-0-ga916309 Expected:v1.1.2-0-ga916309} InitCommit:{ID:de40ad0 Expected:de40ad0} SecurityOptions:[name=apparmor name=seccomp,profile=default] ProductLicense: Warnings:<nil> ServerErrors:[] ClientInfo:{Debug:false Plugins:[map[Experimental:true Name:app Path:/usr/libexec/docker/cli-plugins/docker-app SchemaVersion:0.1.0 ShortDescription:Docker App Vendor:Docker Inc. Version:v0.9.1-beta3] map[Name:buildx Path:/usr/libexec/docker/cli-plugins/docker-buildx SchemaVersion:0.1.0 ShortDescription:Docker Buildx Vendor:Docker Inc. Version:v0.8.2-docker] map[Name:compose Path:/usr/libexec/docker/cli-plugins/docker-compose SchemaVersion:0.1.0 ShortDescription:Docker Compose Vendor:Docker Inc. Version:v2.6.0] map[Name:scan Path:/usr/libexec/docker/cli-plugins/docker-scan SchemaVersion:0.1.0 ShortDescription:Docker Scan Vendor:Docker Inc. Version:v0.17.0]] Warnings:<nil>}}
I0803 02:36:33.487683   16878 api_server.go:165] Checking apiserver status ...
I0803 02:36:33.487790   16878 ssh_runner.go:195] Run: sudo pgrep -xnf kube-apiserver.*minikube.*
I0803 02:36:33.487882   16878 cli_runner.go:164] Run: docker container inspect -f "'{{(index (index .NetworkSettings.Ports "22/tcp") 0).HostPort}}'" granular
I0803 02:36:33.527614   16878 sshutil.go:53] new ssh client: &{IP:127.0.0.1 Port:49157 SSHKeyPath:/home/akash/.minikube/machines/granular/id_rsa Username:docker}
I0803 02:36:33.690187   16878 ssh_runner.go:195] Run: sudo egrep ^[0-9]+:freezer: /proc/2320/cgroup
I0803 02:36:33.708873   16878 api_server.go:181] apiserver freezer: "11:freezer:/docker/d3d93fc9839a93e87855faf0bf57396366b150f34a13b3f1fcc30197ded2f6e6/kubepods/burstable/pod59496dffdcbd4cde1761d910d58b318b/5091048e07eac79ddde32226248317a2f2dd95886907b5d04c25c33429980b64"
I0803 02:36:33.708988   16878 ssh_runner.go:195] Run: sudo cat /sys/fs/cgroup/freezer/docker/d3d93fc9839a93e87855faf0bf57396366b150f34a13b3f1fcc30197ded2f6e6/kubepods/burstable/pod59496dffdcbd4cde1761d910d58b318b/5091048e07eac79ddde32226248317a2f2dd95886907b5d04c25c33429980b64/freezer.state
I0803 02:36:33.718749   16878 api_server.go:203] freezer state: "THAWED"
I0803 02:36:33.718790   16878 api_server.go:240] Checking apiserver healthz at https://192.168.49.2:8443/healthz ...
I0803 02:36:33.728111   16878 api_server.go:266] https://192.168.49.2:8443/healthz returned 200:
ok
I0803 02:36:33.728196   16878 tunnel.go:65] Checking for tunnels to cleanup...
I0803 02:36:33.731641   16878 host.go:66] Checking if "granular" exists ...
I0803 02:36:33.732008   16878 cli_runner.go:164] Run: docker container inspect granular --format={{.State.Status}}
I0803 02:36:33.770195   16878 cli_runner.go:164] Run: docker container inspect -f "{{range .NetworkSettings.Networks}}{{.IPAddress}},{{.GlobalIPv6Address}}{{end}}" granular
I0803 02:36:33.812049   16878 tunnel_manager.go:71] Setting up tunnel...
I0803 02:36:33.812083   16878 tunnel_manager.go:81] Started minikube tunnel.
I0803 02:36:38.812470   16878 host.go:66] Checking if "granular" exists ...
I0803 02:36:38.813119   16878 cli_runner.go:164] Run: docker container inspect granular --format={{.State.Status}}
I0803 02:36:38.856961   16878 route_linux.go:40] Adding route for CIDR 10.96.0.0/12 to gateway 192.168.49.2
I0803 02:36:38.857040   16878 route_linux.go:42] About to run command: [sudo ip route add 10.96.0.0/12 via 192.168.49.2]
[sudo] password for akash: 
I0803 02:36:54.427454   16878 route_linux.go:48] []
I0803 02:36:54.444581   16878 loadbalancer_patcher.go:80] callisto-ui-service is type LoadBalancer.
I0803 02:36:54.452247   16878 loadbalancer_patcher.go:122] Patched callisto-ui-service with IP 10.102.52.59
I0803 02:36:54.452289   16878 loadbalancer_patcher.go:80] kong-kong-proxy is type LoadBalancer.
I0803 02:36:54.458125   16878 loadbalancer_patcher.go:122] Patched kong-kong-proxy with IP 10.97.79.54
I0803 02:36:54.458164   16878 loadbalancer_patcher.go:80] luna-ui-service is type LoadBalancer.
I0803 02:36:54.463922   16878 loadbalancer_patcher.go:122] Patched luna-ui-service with IP 10.98.137.82

### Operating System

Ubuntu

### Driver

Docker

**Comment by klaases** [comment]
Hi @hal-shin, please make sure you are on the latest version of minikube for https://github.com/kubernetes/minikube/pull/14245 to take effect, see v.1.26.1.

Also, thank you for sharing your experience! If you don't mind, could you provide minikube logs?

Please attach (drag) the`logs.txt` file to this issue  which can be generated by running this command:

```
$ minikube logs --file=logs.txt
```` 

This will help us isolate the problem further. Thank you!

/triage needs-information
/kind support

**Comment by hal-shin** [comment]
Hi @klaases, sorry for the delay, the initial log was on a coworker's computer as I didn't have a linux system. I've since built my own to debug this, and I'm seeing the same issue.

Minikube version is v1.26.1, but I believe the `--bind-address` flag was introduced in v1.26.0.

[logs.txt](https://github.com/kubernetes/minikube/files/9355233/logs.txt)

**Comment by yuval-illumex** [comment]
Hi @klaases , any chance to take a look on this one?
I'm having the same issue as well.
I need to bind address to 127.0.0.1 to use 3rd party oauth server.

You need me to attach logs as well?

**Comment by klaases** [comment]
Hi @hal-shin, is it possible that there is another service running on the system that could be causing the port to be unavailable?

```
E0816 22:41:16.232174       1 proxier.go:1600] "can't open port, skipping it" err="listen tcp4 :30279: bind: address already in use" port={Description:nodePort for callisto/callisto-ui-service IP: IPFamily:4 Port:30279 Protocol:TCP}
```

Hi @yuval-illumex, are you also seeing an error similar to the above?

**Comment by hal-shin** [comment]
@klaases I can confirm that there are no conflicting ports with my host machine.

**Comment by pmahony893** [comment]
From reading through the code, it appears that the `bind-address` option is only used if the code determines that port forwarding is needed, and Kubernetes is running in a container. This condition evaluates to false on a Linux host running docker locally as root.

Relevant logic is in:
- https://github.com/kubernetes/minikube/blob/5fb961c517f45cf2f1f19e51f44dc9495d319b76/cmd/minikube/cmd/tunnel.go#L88
- https://github.com/kubernetes/minikube/blob/3e839c44001c432fef479a132cfd9b5200032d98/pkg/minikube/driver/driver.go#L199

**Comment by spowelljr** [comment]
Good find @pmahony893, I'm assuming the logic should be updated to `NeedsPortForward` or `bindAddress` is set.

**Comment by spowelljr** [comment]
@hal-shin @pmahony893 @yuval-illumex 

Could you try the binary of the PR and see if it resolves your issue

https://storage.googleapis.com/minikube-builds/14951/minikube-linux-amd64

**Comment by hal-shin** [comment]
@spowelljr Hey, sorry for the delay, but I've just tried your binary and it works great! Thank you so much for this fix!

**Comment by spowelljr** [comment]
Glad it works @hal-shin, it will be included in the v1.27.1 release which should be released Thursday

**Comment by anton21m** [comment]
I ansmer here: https://stackoverflow.com/questions/58561682/minikube-with-ingress-example-not-working/76982013#76982013
For solution it's problem
